nwbwidgets/spectrum.py

```diff
@@ -0,0 +1,59 @@
+from ndx_spectrum import Spectrum
+import warnings
+import matplotlib.pyplot as plt
+from collections.abc import Iterable
+from h5py import Dataset
+from ipywidgets import widgets
+import numpy as np
+from ipywidgets.widgets.interaction import show_inline_matplotlib_plots
+
+
+def show_spectrum(node, **kwargs):
+    if isinstance(node, Spectrum):
```

Comment:
do type checking at the beginning of the function. Should be `assert isinstance(node, Spectrum), 'node must be of type Spectrum'`

Reply:
I wanted to raise only a warning if it fails

Reply:
ok then do that first

Reply:
This is a function that requires a certain data type. If it is not given that data type, it should throw an appropriate error, not return None

Reply:
I'd still like you to do this. Cutting out early allows you to un-indent the entire rest of the function.

Reply:
okay makes sense, fixed https://github.com/NeurodataWithoutBorders/nwb-jupyter-widgets/pull/127/commits/3b9f7357ecccfbba24fa8fc10d42c2f81c205a6c